pkg/controllers/management/clusterdeploy/clusterdeploy.go

```diff
@@ -403,6 +412,29 @@ func (cd *clusterDeploy) setNetworkPolicyAnn(cluster *v3.Cluster) error {
 	return nil
 }
 
+// restartAgent performs a kubectl rollout restart on the cluster agent deployment
+// if a force restart was requested and if it wasn't already restarted by kubectl apply.
+func (cd *clusterDeploy) restartAgent(originalResourceVersion string, cluster *v3.Cluster) error {
+	kubeConfig, _, err := cd.getKubeConfig(cluster)
+	if err != nil {
+		return err
+	}
+	if cluster.Annotations[AgentForceRestartAnn] != "true" {
+		return nil
+	}
+	if originalResourceVersion != cd.getClusterAgentResourceVersion(cluster.Name) {
+		// kubectl apply should have already restarted this deployment
+		return nil
+	}
```

Comment:
Couldn't the resource version be bumped for other reasons? I'm not sure if this is enough to infer this.

Reply:
Would another option be to put the annotation on the agent deployment before updating the cluster, that way it will be removed once the agent is redeployed?

Reply:
Not sure I parse this, are you saying to put an annotation on the downstream agent deployment resource instead of the local cluster resource, in order to force the agent to be redeployed?